docs/Specification.md

```diff
@@ -173,6 +176,43 @@ The salt is always `covertpassphrase` (16 bytes)
 
 Notice that in this stage the roles of salt and password are reversed because libsodium requires a salt of exactly 16 bytes, matching the pwhash but not the 12-byte nonce.
 
+## Covert Double Ratchet
+
+The ratchet mode enables conversations with **forward secrecy** and self-deleting message keys such that old messages cannot be read later even if one or both parties' keys are breached and the attacker has access to all ciphertext. Further, it provides **post-breach secrecy** such that even after such breach new keys are negotiated within a message round-trip such that no further communication is compromised. Messages may be received out of order or some of them may be lost but the recipient can recover the correct ordering and detect any prior missing messages.
+
+The documentation of ratchet operation is somewhat incomplete, and for implementation you will need to consult Covert source code and Signal's [Double Ratchet with header encryption](https://signal.org/docs/specifications/doubleratchet/#double-ratchet-with-header-encryption)
+
+### Initial messages using public keys
+
+The sender of a public-key encrypted message may signal his capability to support Double Ratchet in replies by adding an `r: N` field in the archive index header. This should only be used when there is a single recipient, as the double ratchet only functions between two parties. Further, both parties will need to use a persistent **idstore** to keep track of the ratchet keys in use. The sender stores locally the filehash of each message sent, which becomes the header encryption key for the initial ratchet reply.
```

Comment:
Since this is the first time "N" is mentioned, it should also be explained. This would be message number stating from 0, 1, ... presumably.